sdk/python/requirements.in

```diff
@@ -39,6 +40,3 @@ uritemplate>=3.0.1,<4
 
 ## kfp.deprecated.cli ##
 typer>=0.3.2,<1.0
-
-## standard library backports ##
-typing-extensions>=3.7.4,<5; python_version<"3.9"
```

Comment:
Curious are there new features used so that we want this package even for Python 3.9+?

Reply:
I added this because I used it for tests in `custom_artifact_types_test.py`. Since it's only a test dependency we have the option include in the `presubmit-tests-sdk.sh` installs only, but I suspect this will trip up local developers who do not install the `dev-requirements.txt` file.

I'm in favor of requiring this dependency with `kfp` for all Python versions as this seems like a low risk dependency with itself 0 additional dependencies. WDYT?

Reply:
I thought this wasn't a test-only dependency. But sounds good to me to include it for all Python version.

Reply:
Opted to refactor and incidentally removed the need for this